### PR TITLE
Add CupertinoSlider

### PR DESCRIPTION
**Pull request by abarth**
This widget matches the style of the iOS slider widget. We don't yet
have the proper visual design for the disabled state.

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie @HansMuller

File: packages/flutter/test/cupertino/slider_test.dart [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+                  });
+                },
+              ),
+            ),
+          );
+        },
+      ),
+    );
+
+    expect(value, equals(0.0));
```
Comment (Hixie) [inline packages/flutter/test/cupertino/slider_test.dart]: This test should verify in more detail that you don't have the iOS bug

File: packages/flutter/lib/src/cupertino/slider.dart [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+import 'dart:ui' show lerpDouble;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/gestures.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/widgets.dart';
+
+import 'thumb_painter.dart';
+
+/// A iOS-style slider.
```
Comment (Hixie) [inline packages/flutter/lib/src/cupertino/slider.dart]: nit: An

File: packages/flutter/lib/src/cupertino/slider.dart [hunk 3]
```diff
@@ -0,0 +1,345 @@
...
+    if (isInteractive)
+      semantics.addAdjustmentActions();
+  }
+
+  @override
+  void performAction(SemanticsAction action) {
+    switch (action) {
+      case SemanticsAction.increase:
+        if (isInteractive)
+          onChanged((value + _kAdjustmentUnit).clamp(0.0, 1.0));
```
Comment (Hixie) [inline packages/flutter/lib/src/cupertino/slider.dart]: for discrete ones, should we move by the discrete unit?

File: packages/flutter/lib/src/cupertino/thumb_painter.dart [hunk 4]
```diff
@@ -0,0 +1,35 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/painting.dart';
+
+final MaskFilter _kShadowMaskFilter = new MaskFilter.blur(BlurStyle.normal, BoxShadow.convertRadiusToSigma(1.0));
+
+class CupertinoThumbPainter {
```
Comment (Hixie) [inline packages/flutter/lib/src/cupertino/thumb_painter.dart]: Having this as a class like this is fine. I wouldn't bother making it a decoration for now.

File: packages/flutter/test/cupertino/slider_test.dart [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    final Point topLeft = tester.getTopLeft(find.byKey(sliderKey));
+    const double unit = CupertinoThumbPainter.radius;
+    await tester.scrollAt(
+        topLeft + const Offset(unit, unit), const Offset(3.0 * unit, 0.0));
+    expect(value, greaterThan(0.0));
+    expect(value, lessThan(0.5));
+    await tester.pump(); // No animation should start.
+    expect(SchedulerBinding.instance.transientCallbackCount, equals(0));
```
Comment (Hixie) [inline packages/flutter/test/cupertino/slider_test.dart]: pretty sure the test framework does this for you automatically, fwiw

**Comment by Hixie** [comment]
LGTM

**Pull request merged**